app/services/data_migrations/fix_lat_long_flip_flops.rb

```diff
@@ -0,0 +1,36 @@
+module DataMigrations
+  class FixLatLongFlipFlops
+    TIMESTAMP = 20210323203521
+    MANUAL_RUN = false
+
+    def change
+      audits = Audited::Audit.where(
+        "auditable_type = 'Provider' AND
+        action = 'update' AND
+        audited_changes ?& array['latitude', 'longitude'] AND
+        (audited_changes - 'latitude') - 'longitude' = '{}'",
+      ).order(created_at: :asc)
+
+      audit_count = 0
+      total_audits = audits.count
+
+      Provider.find_each do |p|
+        audits_for_provider = audits.where(auditable_id: p.id)
+
+        # if we already had lat/lng when we made the first spurious update, delete every audit
+        if audits.first.audited_changes['latitude'].last.nil?
+          audits_for_provider.destroy_all
+
+        # otherwise, delete all the audits except the one that set it for the first time
+        else
+          audits_for_provider.where("audited_changes#>>'{longitude, 1}' is null").destroy_all
+          audits_for_provider.where("audited_changes#>>'{longitude, 0}' is null").offset(1).destroy_all
+        end
```

Comment:
If the last flip was setting the values to `null` the data would be in a inconsistent state with the audit trail. I guess the data itself would get fixed up the next time the sync happens so it's not a big problem.

Reply:
I'm not sure the `is null` condition works as intended here. The problem is that `audited_changes#>>'{longitude, 0}'` is `null` for records that don't have the `longitude` key at all.

Maybe something like `audited_changes#>>'{longitude, 1}' is not null AND audited_changes#>>'{longitude, 0}' is null` etc. is needed.

Reply:
I _think_ the original query guards against that by checking that the audited_changes includes both latitude and longitude and it also contains no other keys?

Reply:
Wouldn't line 26 delete all flips to null?

Reply:
Ah yes I think you're right about the conditions, that's makes sense now!

For the data consistency thing, my only concern was that the lat/lng values would be null immediately after the migration but that's dumb - we fixed the flips to null a while back so all the current data should be set to non-null values (where they ever existed).